src/app/utils/auth/socialSignInHelpers.ts

```
@@ -0,0 +1,130 @@
+    return
+  } else {
+    captureMessage("AUTH_FAILURE: " + error.message)
+
+    const canBeLinked =
+      error.error === "User Already Exists" && error.meta && error.meta.existingProviders
+    if (canBeLinked) {
+      handleErrorWithAlternativeProviders(error.meta)
+      return
+    }
+    GlobalStore.actions.auth.setSessionState({ isLoading: false })
+
+    InteractionManager.runAfterInteractions(() => {
+      const errorMode = error.message === "Attempt blocked" ? "attempt blocked" : "no account"
+      showErrorAlert(errorMode, error)
+      return
+    })
+  }
+}
+
+const handleErrorWithAlternativeProviders = (meta: AuthPromiseRejectType["meta"]) => {
+  if (!meta) {
+    return
+  }
+
+  const titleizedProvider = capitalize(meta.provider)
+
+  const {
+    email,
+    name,
+    existingProviders: providers,
+    provider: providerToBeLinked,
+    oauthToken,
+    idToken,
+    appleUid,
+  } = meta
+
+  const navParams: Omit<
+    OnboardingNavigationStack["OnboardingSocialLink"],
+    "tokenForProviderToBeLinked"
+  > = {
+    email,
+    name: name ?? "",
+    providers: providers ?? [],
+    providerToBeLinked,
+  }
+
+  let tokenForProviderToBeLinked: GoogleOrFacebookToken | AppleToken
+
+  if (["google", "facebook"].includes(providerToBeLinked)) {
+    if (!oauthToken) {
+      console.warn(`Error: No oauthToken provided for ${titleizedProvider}`)
+      return
+    }
+    tokenForProviderToBeLinked = oauthToken
+
+    console.log({ navParams, tokenForProviderToBeLinked })
+
+    // TODO:
+    // navigation.navigate("OnboardingSocialLink", {
+    //   ...navParams,
+    //   tokenForProviderToBeLinked,
+    // })
+  } else if (providerToBeLinked === "apple") {
+    if (!idToken || !appleUid) {
+      console.warn(`Error: idToken and appleUid must be provided for ${titleizedProvider}`)
+      return
+    }
+    tokenForProviderToBeLinked = { idToken, appleUid }
+
+    console.log({ navParams, tokenForProviderToBeLinked })
```

🦑

---

I'm going to continue working on this helper when I get to social account linking (it's ticketed), so in the meantime I decided to console log these variables so that I can suppress errors without commenting-out the whole file.